hack/generator/pkg/codegen/testdata/ArmResource/Arm_test_embedded_resource_multiple_contexts_azure.golden

```diff
@@ -0,0 +1,834 @@
+// Code generated by k8s-infra. DO NOT EDIT.
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+package v1alpha1api20200101
+
+import (
```

Comment:
Whoa this is complicated! It's quite hard to see the resulting properties types through the forest of arm conversions around them (exacerbated by it being in github and the golden file not syntax highlighted, so hard to jump to definitions). 

I think it's correct though - it seems like the crux of it is `FakeResourceProperties_SubResourceEmbedded` vs `FakeResourceProperties_SubResourceEmbedded_1`, right?

What about adding a .doc file alongside the .json and .golden file for this to explain the initial setup and the expected properties types without the noise? Or maybe it would be worth having golden tests for this that didn't include the arm conversions? Not sure how difficult that is though.

Reply:
> it seems like the crux of it is FakeResourceProperties_SubResourceEmbedded vs FakeResourceProperties_SubResourceEmbedded_1, right?

yeah, that's right

> What about adding a .doc file alongside the .json and .golden file for this to explain the initial setup and the expected properties types without the noise? Or maybe it would be worth having golden tests for this that didn't include the arm conversions? Not sure how difficult that is though.

Dropping the ARM conversion stuff isn't that _hard_ but it means that we need to have yet another flavor of test "kind" I think, which I'm not sure really serves us anyway as ideally we'd ensure that ARM types for this situation are shaped correctly too? I don't mind the .doc file idea though for a better formatted description of the test - would you think it would replace the existing description at the top of the JSON or just augment it?

Reply:
I'd say just to augment it.